packages/editor/src/components/post-preview-button/test/index.js

```diff
@@ -1,188 +1,319 @@
 /**
  * External dependencies
  */
-import { shallow } from 'enzyme';
+import { render, screen, within } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
 
 /**
  * Internal dependencies
  */
 import { PostPreviewButton } from '../';
 
 describe( 'PostPreviewButton', () => {
-	describe( 'setPreviewWindowLink()', () => {
-		it( 'should do nothing if there is no preview window', () => {
-			const url = 'https://wordpress.org';
-			const setter = jest.fn();
-			const wrapper = shallow( <PostPreviewButton /> );
+	const documentWrite = jest.fn();
+	const documentTitle = jest.fn();
+	const documentClose = jest.fn();
+	const setLocation = jest.fn();
 
-			wrapper.instance().setPreviewWindowLink( url );
+	beforeEach( () => {
+		global.open = jest.fn( () => ( {
+			focus: jest.fn(),
+			document: {
+				write: documentWrite,
+				close: documentClose,
+				get title() {},
+				set title( value ) {
+					documentTitle( value );
+				},
+			},
+			get location() {},
+			set location( value ) {
+				setLocation( value );
+			},
+		} ) );
+	} );
 
-			expect( setter ).not.toHaveBeenCalled();
-		} );
+	afterEach( () => {
+		global.open.mockRestore();
+		documentWrite.mockReset();
+		documentTitle.mockReset();
+		documentClose.mockReset();
+		setLocation.mockReset();
+	} );
 
-		it( 'set preview window location to url', () => {
-			const url = 'https://wordpress.org';
-			const setter = jest.fn();
-			const wrapper = shallow( <PostPreviewButton /> );
-			wrapper.instance().previewWindow = {
-				get location() {
-					return {
-						href: 'about:blank',
-					};
-				},
-				set location( value ) {
-					setter( value );
-				},
-			};
+	it( 'should render with `editor-post-preview` class if no `className` is specified.', () => {
+		render( <PostPreviewButton /> );
 
-			wrapper.instance().setPreviewWindowLink( url );
+		expect( screen.getByRole( 'button' ) ).toHaveClass(
+			'editor-post-preview'
+		);
+	} );
 
-			expect( setter ).toHaveBeenCalledWith( url );
-		} );
+	it( 'should render with a custom class and not `editor-post-preview` if `className` is specified.', () => {
+		render( <PostPreviewButton className="foo-bar" /> );
+
+		const button = screen.getByRole( 'button' );
+
+		expect( button ).toHaveClass( 'foo-bar' );
+		expect( button ).not.toHaveClass( 'editor-post-preview' );
 	} );
 
-	describe( 'getWindowTarget()', () => {
-		it( 'returns a string unique to the post id', () => {
-			const instance = new PostPreviewButton( {
-				postId: 1,
-			} );
+	it( 'should set `variant` to `tertiary` if no classname is specified.', () => {
```

Comment:
Nit: to avoid exposing `Button`-related implementation details, a better name could be

```suggestion
	it( 'should render a tertiary button if no classname is specified.', () => {
```